lib/screens/notifications/widgets/connection_request_response.dart

```diff
@@ -1,15 +1,33 @@
 import 'package:flutter/material.dart';
 import 'package:junto_beta_mobile/backend/repositories.dart';
 import 'package:junto_beta_mobile/backend/backend.dart';
+import 'package:junto_beta_mobile/models/models.dart';
 import 'package:junto_beta_mobile/screens/notifications/widgets/request_response_button.dart';
 import 'package:junto_beta_mobile/screens/notifications/notifications_handler.dart';
 import 'package:junto_beta_mobile/utils/junto_overlay.dart';
 import 'package:provider/provider.dart';
 
 class ConnectionRequestResponse extends StatelessWidget {
-  const ConnectionRequestResponse({this.userAddress});
+  const ConnectionRequestResponse({this.userAddress, this.notification});
 
   final String userAddress;
+  final JuntoNotification notification;
+
+  void _respondToRequest(BuildContext context, bool response) async {
+    // show Junto loader
+    JuntoLoader.showLoader(context);
+    // respond to request
+    await Provider.of<UserRepo>(context, listen: false)
+        .respondToConnection(userAddress, response);
+    // delete notification from cache
+    await Provider.of<LocalCache>(context, listen: false)
+        .deleteNotification(notification.address);
+    // refetch notifications
```

Comment:
Don't access notifications through the `LocalCache`. Do it through `NotificationsHandler` as `NotificationsHandler` stores a list of current notifications in memory. Removing it from cache won't remove it from UI. 

Also, don't access repositories/cache in widgets directly.

Reply:
Deleting it via LocalCache produces the desired result. I tried it via NotificationsHandler and it doesn't get removed from the UI. The issue is that the notifications are still being persisted in the Hive cache, despite being deleted from the API is it not? If so, can you clarify why the LocalCache approach isn't correct?

Reply:
Don't access LocalCache directly in widgets, but expose operations on notifications via NotificationsHandler. Of course the final result should be to remove notifications from cache, but do it through NotificationsHandler similarly to how notifications are marked as read. This way we won't have to touch widgets code if we want to modify how are we caching notifications.